src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/UseLiteralsWhereAppropriate.Fixer.cs

```diff
@@ -20,11 +26,84 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
-            
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode declaration = root.FindNode(context.Span);
+            declaration = SyntaxGenerator.GetGenerator(context.Document).GetDeclaration(declaration, DeclarationKind.Field);
+            var fieldFeclaration = GetFieldDeclaration(declaration);
+            if (fieldFeclaration == null)
+            {
+                return;
+            }
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftQualityGuidelinesAnalyzersResources.UseLiteralsWhereAppropriateCodeActionTitle,
+                    cancellationToken => ToConstantDeclarationAsync(context.Document, fieldFeclaration, cancellationToken)),
+                diagnostic);
+        }
+
+        private async Task<Document> ToConstantDeclarationAsync(Document document, SyntaxNode fieldDeclaration, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+
+            SyntaxTriviaList leadingTrivia = new SyntaxTriviaList();
+            SyntaxTriviaList trailingTrivia = new SyntaxTriviaList();
+
+            SyntaxTokenList newModifiers = new SyntaxTokenList();
+            foreach (SyntaxToken modifier in GetModifiers(fieldDeclaration))
+            {
+                if (IsStaticKeyword(modifier) || IsReadonlyKeyword(modifier))
+                {
+                    if (leadingTrivia.Count == 0 && trailingTrivia.Count == 0)
+                    {
+                        leadingTrivia = leadingTrivia.AddRange(modifier.LeadingTrivia);
+                        trailingTrivia = trailingTrivia.AddRange(modifier.TrailingTrivia);
+                    }
+                    else
+                    {
+                        trailingTrivia = trailingTrivia.AddRange(modifier.LeadingTrivia);
+                        trailingTrivia = trailingTrivia.AddRange(modifier.TrailingTrivia);
+
+                        var constModifier =
```

Comment:
Nope. As explained above, entering the inner `else` clause means we have processed both the keywords `static` and `readonly`. That's the place to insert. In case any additional modifiers will follow, their relative position should not change.